src/Raven.Client/Documents/Conventions/DocumentConventions.cs

```diff
@@ -252,6 +254,16 @@ public DocumentConventions()
         private Size _maxContextSizeToKeep;
         private ISerializationConventions _serialization;
 
+        public EventHandler<ShouldIgnoreEntityChangesEventArgs> ShouldIgnoreEntityChanges
```

Comment:
That is the first time we use `EventHandler / EventArgs` in our conventions. Why not to change to `Func` returning `bool`?

Reply:
There is a recommendation against returning bool from events. Because you may have multiple subscribers, and only the last one will be respected.

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/event-pattern#returning-values-from-event-subscribers

Reply:
That said, we are returning values from other events here

Reply:
Hmm.. okay, but the issue is that this is something totally different from the rest of our API.

Does it make sense to have multiple subscribers here? I think we were always assuming that there will be only one. Maybe we should explicitly forbid multiple subscribers and throw then.